extract-method-core/src/main/java/org/jetbrains/research/extractMethod/core/extractors/CustomRefactoringHandler.java

```diff
@@ -14,41 +14,56 @@
 import gr.uom.java.xmi.diff.ExtractOperationRefactoring;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.jetbrains.research.extractMethod.metrics.MetricCalculator;
+import org.jetbrains.research.extractMethod.metrics.features.FeaturesVector;
+import org.jetbrains.research.extractMethod.metrics.location.LocationVector;
+import org.jetbrains.research.extractMethod.metrics.utils.DatasetRecord;
 import org.refactoringminer.api.Refactoring;
 import org.refactoringminer.api.RefactoringHandler;
 import org.refactoringminer.api.RefactoringType;
 
 import java.io.FileWriter;
 import java.io.IOException;
-import java.nio.file.Path;
-import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.stream.Collectors;
 
+import static org.jetbrains.research.extractMethod.core.utils.LocationUtil.buildLocationVector;
 import static org.jetbrains.research.extractMethod.core.utils.PsiUtil.findMethodBySignature;
 import static org.jetbrains.research.extractMethod.core.utils.StringUtil.calculateSignature;
-import static org.jetbrains.research.extractMethod.metrics.MetricCalculator.writeFeaturesToFile;
 
 public class CustomRefactoringHandler extends RefactoringHandler {
     private final Project project;
     private final GitCommit gitCommit;
-    private final String repositoryPath;
+    private final String repoFullName;
+    private final String priorCommitHash;
     private final FileWriter fileWriter;
     private final Logger LOG = LogManager.getLogger(CustomRefactoringHandler.class);
 
     public CustomRefactoringHandler(Project project,
-                                    String repositoryPath,
+                                    String repoFullName,
                                     GitCommit gitCommit,
+                                    String priorCommitHash,
                                     FileWriter fileWriter) {
         this.project = project;
-        this.repositoryPath = repositoryPath;
+        this.repoFullName = repoFullName;
         this.gitCommit = gitCommit;
+        this.priorCommitHash = priorCommitHash;
         this.fileWriter = fileWriter;
     }
 
+    private static String getMethodSlice(PsiFile psiFile, int beginLine, int endLine) {
+        String[] fileLines = psiFile.getText().split("\n");
+        List<String> resultingLines = new ArrayList<>();
+        for (int i = 0; i < fileLines.length; i++)
+            if (i + 1 >= beginLine && i + 1 <= endLine)
```

Comment:
does beginLine start with 0 or 1?
please, test the case when endLine is equal to method lines size

Reply:
both beginLine and endLine are literal number of lines (starting from 1), where the first line is the first line in file. E.g. line with `for` statement has beginLine=60, endLine=60. The method works correctly (where applicable) based on my experience